packages/main/src/plugin/kubernetes/kubernetes-client.ts

```diff
@@ -403,6 +403,35 @@ export class KubernetesClient {
     this.apiSender.send('kubernetes-context-update');
   }
 
+  async updateContext(contextName: string, newContextName: string, newContextNamespace: string): Promise<void> {
+    const newConfig = new KubeConfig();
+
+    const originalContext = this.kubeConfig.contexts.find(context => context.name === contextName);
+    const newContexts = this.kubeConfig.contexts.filter(ctx => ctx.name !== contextName);
+    if (!originalContext) return;
```

Comment:
You may want to raise an error instead, and you can do it before computing `newContexts`